ui/imports/shared/views/chat/InvitationBubbleView.qml

```diff
@@ -280,36 +280,49 @@ Item {
+                    clip: true
+                    StatusFlatButton {
+                        id: joinBtn
+                        width: parent.width
+                        height: (parent.height+Style.current.padding)
+                        anchors.top: parent.top
+                        anchors.topMargin: -Style.current.padding
+                        text: qsTr("Unsupported state")
+                        contentItem: Item {
+                            StatusBaseText {
+                                anchors.centerIn: parent
+                                anchors.verticalCenterOffset: Style.current.halfPadding
+                                font: joinBtn.font
+                                color: joinBtn.enabled ? joinBtn.textColor : joinBtn.disabledTextColor
+                                text: joinBtn.text
+                            }
                         }
-                        else if (rectangleBubble.state === "requestToJoin") {
-                            Global.openPopup(communityIntroDialog, { joinMethod: () => {
-                                                        let error = root.store.requestToJoinCommunity(communityId, userProfile.name)
-                                                        if (error) joiningError.showError(error)
-                                                    } });
+
+                        onClicked: {
+                            if (rectangleBubble.state === "joined") {
+                                root.store.setActiveCommunity(communityId);
+                                return
+                            }
+                            if (rectangleBubble.state === "unjoined") {
+                                Global.openPopup(communityIntroDialog, { joinMethod: () => {
+                                                         let error = root.store.joinCommunity(communityId, userProfile.name)
+                                                         if (error) joiningError.showError(error)
+                                                     } });
+                            }
+                            else if (rectangleBubble.state === "requestToJoin") {
+                                Global.openPopup(communityIntroDialog, { joinMethod: () => {
+                                                         let error = root.store.requestToJoinCommunity(communityId, userProfile.name)
+                                                         if (error) joiningError.showError(error)
+                                                     } });
+                            }
                         }
-                    }
 
-                    Component.onCompleted: {
-                        // FIXME: extract StatusButtonBackground or expose radius property in StatusBaseButton
-                        background.radius = 16
+                        Component.onCompleted: {
+                            background.radius = Style.current.padding;
```

Comment:
Can't this be set right on the background as a binding?

Reply:
you mean like `background.radius: Style.current.padding`? unfortunately it doesn't even start, `qrc:/imports/shared/views/chat/InvitationBubbleView.qml:322:36: Cannot assign to non-existent property "radius"`

Reply:
Ah, `background` is an `Item` :/ Ignore me